### PR TITLE
Insight analytics fixes

### LINKED ISSUE #7347: Incorrect reporting on Insights
## Bug description / Expected behavior

While exploring the app today, I found some important issues with reporting on insights:
1. When an insight is viewed from the "View mode" (i.e. when accessed by clicking on a saved insight) we don't trigger "insight viewed" or "insight analyzed". In the event, we should include an attribute that indicates the view mode too.
2. When opening an insight (funnel or otherwise) from saved insights into "View mode" we trigger `correlation viewed` & `correlation properties viewed` automatically even though correlation isn't actually viewed. CC @neilkakkar 
3. We don't report when a user has "compare previous" enabled

## How to reproduce

1. Open an insight from the saved insights list with the JS console open. See events triggered.

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog (ClickHouse-based), version/commit: 1.30.0
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Resolves #7347. All three points should be fixed. I also instrumented a new event: "insight created".

## How did you test this code?

Reproduced cases mentioned in the issue manually. Unfortunately this isn't really testable with Jest, as the instrumentations relies heavily on how `insightLogic` is used, meaning any refactor of insight components and other logics could easily make tests test the wrong thing. Some E2E testing of instrumentation would be ideal, but I don't know if there's a way to do that already and didn't want to get into it currently.

### CODE REVIEW

File: frontend/src/scenes/insights/insightLogic.tsx [hunk 1]
```diff
@@ -306,8 +306,6 @@ export const insightLogic = kea<insightLogicType>({
                 setFilters: (state, { filters }) => cleanFilters(filters, state),
                 setInsight: (state, { insight: { filters }, options: { overrideFilter } }) =>
                     overrideFilter ? cleanFilters(filters || {}) : state,
-                loadInsightSuccess: (state, { insight }) =>
```
Comment (paolodamico) [inline frontend/src/scenes/insights/insightLogic.tsx]: FYI I'm reverting this as this seems to be causing unintended side effects (see failing tests).

**Review by paolodamico — APPROVED** [review]
lgtm! tested on Postgres mode and all 3 cases seem to have been addressed. Thanks @Twixes!

File: frontend/src/scenes/insights/insightLogic.tsx (hunk 1 above)
Comment (Twixes) [reply frontend/src/scenes/insights/insightLogic.tsx]: :+1: I also have some fixes on top of this, I'll push that too

Comment (Twixes) [reply frontend/src/scenes/insights/insightLogic.tsx]: I wanted to add Jest tests for `reportInsightViewed` and the right actions WERE fired according to the `At timeout had received these actions:` list, but for whatever reason the Kea testing scaffolding was not matching the action in the list to the action in `toDispatchActions`... Anyway, refactored things a bit, should be good now

**Pull request merged**